includes/classes/order_total.php

```diff
@@ -160,14 +160,21 @@ function update_credit_account($i) {
   function collect_posts() {
     if (MODULE_ORDER_TOTAL_INSTALLED) {
       reset($this->modules);
+      $collect_posts_rc = true; 
       while (list(, $value) = each($this->modules)) {
         $class = substr($value, 0, strrpos($value, '.'));
         if ( $GLOBALS[$class]->credit_class ) {
           $post_var = 'c' . $GLOBALS[$class]->code;
           if ($_POST[$post_var]) $_SESSION[$post_var] = $_POST[$post_var];
-          $GLOBALS[$class]->collect_posts();
+          $rc = $GLOBALS[$class]->collect_posts();
+          if (isset($rc) && $rc != true) { 
+               $collect_posts_rc = false; 
+          }
         }
       }
+      if (!$collect_posts_rc) { 
+            zen_redirect(zen_href_link(FILENAME_CHECKOUT_PAYMENT, '', 'SSL', true, false));
+      }

```

Comment:
I'm guessing, but does your use of `rc` mean "return code"?

I think semantically you mean to track whether any error occurred when processing `collect_posts()` in the specified module.
In that context would it make the code more readable if `$collect_posts_rc` was `$collect_posts_success` and `$rc` was `$no_errors`?

Also, these changes to `order_total.php` will affect *all* order_total modules that have a `collect_posts()` method, but only `ot_coupon` has been updated in this PR to do set these return values. Shouldn't `ot_gv` also be updated similarly?

NOTE: `ot_gv` does its zen_redirect to CHECKOUT_FLOW instead of CHECKOUT_PAYMENT. I suspect `ot_coupon` should have been using the same CHECKOUT_FLOW redirect. Fortunately you've moved it here to order_total, but probably should update it to redirect to `FILENAME_CHECKOUT_FLOW step=payment`.